### DISCUSSION TITLE
Missing Feature Request

### DISCUSSION THREAD

**Discussion by ICodeDiffusion** [vladmandic/sdnext#989] [Ideas]
So there was a feature you guys had where i can activate an option to use the weights in "()" can be having the "(prompt:1.1)" and it would work with prompts that already had it but only read the "()". Any idea where that setting went? It said emphasis as part of the description of the settings. It can no longer be found when using the browser's find feature. I use this feature on old prompts that use the ":1.1" weights so I don't have to keep hand-editing the prompts when i load up images from the other A1111.

**Comment by EfourC** [comment]
Can you give a couple example prompts of what you mean?

I just tried the following, and they give the same output, meaning the weight for `bird` is calculated the same:
`test (bird) (cat:1.0), forest, sky`
`test (bird:1.1) (cat:1.0), forest, sky`

**Edit:**  Do you mean there were some old prompts that would have `(apple:1.5)`, and the token was literally for 'apple:1.5' with weight calculated at 1.1 by the ( )?

**Reply by brknsoul** [reply]
Sub-question, slight hi-jack: so (prompt:1.3) doesn't work like that anymore?

**Reply by EfourC** [reply]
It does work, I'm just trying to understand the OP's issue.

**Reply by brknsoul** [reply]
Phew. ;-)

**Reply by ICodeDiffusion** [reply]
> Can you give a couple example prompts of what you mean?
> 
> I just tried the following, and they give the same output, meaning the weight for `bird` is calculated the same: `test (bird) (cat:1.0), forest, sky` `test (bird:1.1) (cat:1.0), forest, sky`
> 
> **Edit:** Do you mean there were some old prompts that would have `(apple:1.5)`, and the token was literally for 'apple:1.5' with weight calculated at 1.1 by the ( )?

the edit i believe so, so i can have it just read the "()" in (bird:1.1) so it would read it as (bird).
I want to have it so it automatically ignore the ":1.1" part when i send the png info to the other tabs in the prompts. So i don't have to keep going through each one to remove the ":1.1"
I like using both and switch between them since i can have more options.

**Reply by EfourC** [reply]
I see there are a couple options with 'emphasis' in A1111's current release, but Vlad has removed them from this repo.

However, it doesn't seem like these options would have the effect your describing...  I just tried changing them in A1111 but it didn't remove the ':1.1' syntax out of the (  ). Maybe I was doing something wrong, or A1111 has changed behavior as well?

Screenshot:
![chrome_X3UM0l4adz](https://github.com/vladmandic/automatic/assets/121468033/b5dafc10-c9e3-4af0-9b84-3925b256298d)

**Reply by ICodeDiffusion** [reply]
> I see there are a couple options with 'emphasis' in A1111's current release, but Vlad has removed them from this repo.
> 
> However, it doesn't seem like these options would have the effect your describing... I just tried changing them in A1111 but it didn't remove the ':1.1' syntax out of the ( ). Maybe I was doing something wrong, or A1111 has changed behavior as well?
> 
> Screenshot: ![chrome_X3UM0l4adz](https://user-images.githubusercontent.com/121468033/238556962-b5dafc10-c9e3-4af0-9b84-3925b256298d.png)


"Vlad has removed them" why did he remove them? :/ I really like to use the feature and i hope he bings it back. 
BTW it doesn't remove them it just ignores the :1.1 inputs and only reads the (). Which i use alot for png info so i don't have to keep deleting the :1.1 inputs every time i want to recreate an image. I have specific images that i generate using this feature as well. ( can't seem to recreate the images by removing the :1.1 inputs) and that can probably be the use old karras scheduler that the other A1111(your screenshot variant) has.

**Reply by vladmandic** [reply]
why did i remove them? because it was a legacy feature not maintained or updated for a looong time. only reason it was there since when new emphasis style was added (long time ago), a1111 wasn't sure enough that its not broken, so he kept old one as well.
if someone wants to write a proper alternative prompt parser, i'm all for it, but keeping broken legacy code is not what i want to do. the fact it mostly-worked-for-you-in-specific-case is nice, but not nice enough to keep it.

**Reply by mobilehacker** [reply]
> why did i remove them? because it was a legacy feature not maintained or updated for a looong time. only reason it was there since when new emphasis style was added (long time ago), a1111 wasn't sure enough that its not broken, so he kept old one as well. if someone wants to write a proper alternative prompt parser, i'm all for it, but keeping broken legacy code is not what i want to do. the fact it mostly-worked-for-you-in-specific-case is nice, but not nice enough to keep it.

Clarify for me please, am I doing this right? "a woman in a (((red))) dress" to make the dress red for sure?

**Reply by vladmandic** [reply]
if you run with `--debug`, you can see for yourself:
(short answer - yes, you're adding 1.1x1.1x1.1 to its strenght)

> 11:59:10-848531 DEBUG    Prompt schedule: [[20, 'a woman in a (((red))) dress']]
> 11:59:10-849393 DEBUG    Prompt parse-attention: [['a woman in a', 1.0], ['red', 1.3310000000000004], ['dress', 1.0]]

**Reply by ICodeDiffusion** [reply]
> why did i remove them? because it was a legacy feature not maintained or updated for a looong time. only reason it was there since when new emphasis style was added (long time ago), a1111 wasn't sure enough that its not broken, so he kept old one as well. if someone wants to write a proper alternative prompt parser, i'm all for it, but keeping broken legacy code is not what i want to do. the fact it mostly-worked-for-you-in-specific-case is nice, but not nice enough to keep it.

ok good to know :) Thank you! I'll see if i can make it for myself then haha. I got close to making one thanks to a new extension to chatGPT where i can upload py, js, and c+ files to it and it figured out how to make some working code xD
